beacon-chain/forkchoice/doubly-linked-tree/benchmarks_test.go

```diff
@@ -0,0 +1,52 @@
+package doublylinkedtree
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/testing/require"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+)
+
+func Benchmark_OnBlock(b *testing.B) {
```

Comment:
`OnBlock` is technically not the correct name, it encompasses a lot more things. It sounds like you want to call this insert node and get head or something else